### DISCUSSION TITLE
Geolocation

### DISCUSSION THREAD

**Discussion by noelb** [godotengine/godot-proposals#4187] [Platforms]
Getting a user's geolocation is a core feature of many mobile and web apps. It would be great if Godot had a standardized way to request and access a users geolocation that worked seamlessly across platforms (iOS/android/web).

**Comment by theraot** [comment]
With all the AR/VR improvements the fact that we don't have geolocation/gps is ever more notable.

There was some old third party addon for GPS which no longer works because the way Android plugins work changed. There is newer one which is in portuguese which I managed to load but not to make work (geolocation was never ready). And now Godot 4 is different again, and there is no option as far as I know.

*By the way, I had the never ready issue in Unity, and Apache Cordova.  And I could only solve in Apache Cordova, the solution is to timeout and fallback to `enableHighAccuracy: false`.*

**Reply by theLemon505** [reply]
I tried this in the past for a game idea similar to pokemon go but ran into this issue. It doesnt seem to difficult to apply, just link the engine source to the proper api (google maps, apple maps, etc...). The only hang-up I could think is that since godot is a rather large organization, using google maps or apple maps for free isnt going to be easy and would require alot of management.

**Reply by theraot** [reply]
@theLemon505

Godot organization size won't be an issue. In practice that kind of solution would have the developer of the game provide a key from the appropriate service (it would not be hard coded), and then any payment related to the use of the service would be up to the developer.

By the way, with Apache Cordova I used Leaflet which lets you pick the source of the map tiles (most require a key or account, but at least OpenStreetMaps does not). Something like that would be better for Godot as it would allow it to remain neutral.

**Reply by theraot** [reply]
I should point out that a geolocation solution and map control (i.e. widget) are two different separate things. A good map control should allow us to put any location regardless (from gps or otherwise), and a good geolocation solution should allow us to get the gps location without displaying (in a map or otherwise). And I'm more interested in the geolocation.
